### PR TITLE
Add webxdc caching

### LINKED ISSUE #22: Fix webxdc caching in frontend
When fixing reactivity in 72a4b0315823b1fa49c3964adfde85db4a8dbbf6, I removed appxdc caching because there is no primitive which caches `stores` in solidjs-use. Because of serial based updating, the appstore will not be resend the appinformations, so it will stay empty until new apps are being added.

### LINKED ISSUE #68: Cache downladed apps
currently the app is downloaded every time user click the button, app should be downloaded once, then saved it in localStorage and use the cached version the next time user clicks button, 

only a new download is needed if there is a new version of the app available (would be detected when the appstore.xdc metadata is updated)

### PR DESCRIPTION
**Pull request by Septias**
close #68

### CODE REVIEW

**Comment by adbenitez** [comment]
does this save the files in localStorage or IndexDB? there are limitations for localStorage, check: https://github.com/webxdc/webxdc_docs/issues/56

**Comment by adbenitez** [comment]
> It uses indexdb

great! :rocket:

File: frontend/src/db/shop_db.ts [hunk 1]
```diff
@@ -87,4 +88,28 @@ export class AppInfoDB {
       }
     })
   }
+
+  // Add base64 encoded webxdc to the db.
+  async add_webxdc(webxdc: XDCFile, id: number): Promise<void> {
```
[the remark this answers is not in the document]
Comment (Septias) [reply frontend/src/db/shop_db.ts:93]: Is this a hard issue for you or can we just keep it as is. I  would prefer it the current way

**Review by Simon-Laux — APPROVED** [review]

File: frontend/src/db/shop_db.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (hpk42) [reply frontend/src/db/shop_db.ts:93]: well, i don't understand how the "id" is constructed and how the caching works. there are also no tests which could help to understand it.

**Pull request merged**